cmds/console/i8042_linux.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This file contains support functions for i8042 io for Linux.
+package main
+
+import (
+	"os"
+)
+
+const (
+	Data = 0x60 /* data port */
+
+	Status   = 0x64 /* status port */
+	Inready  = 0x01 /*  input character ready */
+	Outbusy  = 0x02 /*  output busy */
+	Sysflag  = 0x04 /*  system flag */
+	Cmddata  = 0x08 /*  cmd==0 data==1 */
+	Inhibit  = 0x10 /*  keyboard/mouse inhibited */
+	Minready = 0x20 /*  mouse character ready */
+	Rtimeout = 0x40 /*  general timeout */
+	Parity   = 0x80
+
+	Cmd   = 0x64 /* command port (write only) */
+	Nscan = 128
+)
+
+type i8042 struct {
+	data, status int64
+}
+
+func openi8042() (*i8042, error) {
+	var err error
+	f, err = os.OpenFile("/dev/port", os.O_RDWR, 0)
+	if err != nil {
+		return nil, err
+	}
+	return &i8042{data: Data, status: Cmd}, nil
```

Comment:
Right. But `Cmd` is for the writing side of things (speaking for myself only, it helps me to think of them as separate registers). Anyway, it's moot since it's changed.